packages/playground/data-liberation/src/import/WP_Stream_Importer.php

```diff
@@ -395,6 +415,57 @@ private function frontloading_advance_reentrancy_cursor() {
 		}
 	}
 
+	/**
+	 * Sort the entities topologically.
+	 *
+	 * @param int $count The number of entities to process in one go.
+	 */
+	private function topological_sort_next_entity( $count = 10000 ) {
+		if ( null !== $this->next_stage ) {
+			return false;
+		}
+
+		if ( null === $this->entity_iterator ) {
+			$this->entity_iterator    = $this->create_entity_iterator();
+			$this->topological_sorter = new WP_Topological_Sorter( $this->options );
+		}
+
+		if ( ! $this->entity_iterator->valid() ) {
+			$this->entity_iterator  = null;
+			$this->resume_at_entity = null;
+			return false;
+		}
+
+		/**
+		 * Internalize the loop to avoid computing the reentrancy cursor
+		 * on every entity in the imported data stream.
+		 */
+		for ( $i = 0; $i < $count; ++$i ) {
+			if ( ! $this->entity_iterator->valid() ) {
+				break;
+			}
+
+			$entity = $this->entity_iterator->current();
+			$data   = $entity->get_data();
+			$offset = $this->entity_iterator->get_last_xml_byte_offset_outside_of_entity();
```

Comment:
I'm thinking `get_entity_byte_offset()` could be a clearer name for this method

Reply:
Good idea.